common/cluster/metadata.go

```diff
@@ -138,7 +159,6 @@ func (m Metadata) ClusterNameForFailoverVersion(failoverVersion int64) string {
 	if failoverVersion == common.EmptyVersion {
 		return m.currentClusterName
 	}
-
 	initialFailoverVersion := failoverVersion % m.failoverVersionIncrement
 	clusterName, ok := m.versionToClusterName[initialFailoverVersion]
```

Comment:
This doesn't look correct if the domain is currently using minInitialFailoverVersion

Reply:
Good catch 🤦